lib/teams/index.ts

```diff
@@ -214,6 +226,98 @@ export class ApplicationTeam implements Team {
                 hard: this.teamProps.namespaceHardLimits
             }
         });
+        quotaManifest.node.addDependency(this.namespaceManifest);
+    }
+
+    /**
+     * Sets up secrets
+     * @param clusterInfo
+     */
+    protected setupSecrets(clusterInfo: ClusterInfo) {
+        const secrets = this.teamProps.secrets;
+        if (clusterInfo.provisionedAddOns) {
+            if (clusterInfo.provisionedAddOns.has(Constants.SECRETS_STORE_CSI_DRIVER)) {
+                if (Array.isArray(secrets) && secrets.length) {
+                    const cluster = clusterInfo.cluster;
+                    const secretProviderClass = this.teamProps.name + '-aws-secrets';
+                    type secretObject = {
+                        objectName: string;
+                        objectType: SecretType;
+                        region?: string;
+                    }
+                    let objects: secretObject[] = [];
+
+                    const serviceAccount = cluster.addServiceAccount(this.teamProps.name + '-sa', {
+                        name: this.teamProps.name + '-sa',
+                        namespace: this.teamProps.namespace
+                    });
+                    serviceAccount.node.addDependency(this.namespaceManifest);
+
+                    secrets.forEach( (secret) => {
+                        const objectName = secret.secretName;
+                        const objectType = secret.secretType;
+                        const region = secret.secretRegion ? secret.secretRegion : cdk.Aws.REGION;
+                        const accountId = secret.secretAccountId ? secret.secretAccountId : cdk.Aws.ACCOUNT_ID;
+
+                        objects.push({
+                            objectName,
+                            objectType,
+                            region,
+                        });
+
+                        let policyStatement: iam.PolicyStatement;
+
+                        if (secret.secretType === SecretType.SECRETSMANAGER) {
```

Comment:
Consider breaking into distinct methods for manifest generation for readability (keeping method size under ~25 lines unless it is a looong manifest).

Reply:
Done